lib/terraform/group_attributes.rb

```diff
@@ -29,8 +29,8 @@ def self.coerce(config_value)
     end
 
     def resolve(client:)
-      client.each_output_name { |name| soft_store key: name, value: name }
       dup.each_pair { |key, value| store key, client.output(name: value) }
```

Comment:
`#dup` is used because this design is updating the same hash that it's iterating over, as you speculated. I would like to replace this design with one that doesn't alter the original config but that doesn't need to be dealt with in this branch.